src/capi/azext_capi/custom.py

```diff
@@ -364,6 +367,7 @@ def set_azure_identity_secret_env_vars():
 
 
 # pylint: disable=inconsistent-return-statements
+# pylint: disable=too-many-branches
```

Comment:
This is probably because around line 452 we get four or five `if`s deep. Maybe we can refactor this at some point.

Reply:
I agree. I have extracted logic to generate template in its own method